pkg/smokescreen/smokescreen.go

```diff
@@ -475,6 +475,15 @@ func BuildProxy(config *Config) *goproxy.ProxyHttpServer {
 
 		sctx.logger.WithField("url", req.RequestURI).Debug("received HTTP proxy request")
 
+		// verify the request based on the handler, if it exists
+		if config.VerifyRequestHandler != nil {
+			err = config.VerifyRequestHandler(pctx.Req)
```

Comment:
I noticed that we for cleartext requests in this handler we use `req` instead of `pctx.Req`. What's the difference between these two variables? Should we pass `req` here instead for consistency? You can see that above we mutate `req`
```go
req.Header.Del(roleHeader)
req.Header.Del(traceHeader)
```
We'd want to do the same in our handler. I.e. remove extra headers.

Reply:
cc @cds2-stripe in case you have insight.